src/main/java/blip/parser/BlipParser.java

```diff
@@ -0,0 +1,153 @@
+        String[] test = input.split("\\s+", 2);
+        // Missing Event Description.
+        if (test.length < 2 || test[1].equals("")) {
+            throw new EmptyDescriptionException("!!! Missing EVENT Description !!!");
+        }
+        String[] components = test[1].split(" /from | /to ");
+        if (components.length < 3 || components[1].equals("") || components[2].equals("")) {
+            throw new EmptyDescriptionException("!!! Missing EVENT Start/End Date Time !!!");
+        }
+        String description = components[0];
+        String eventStart = components[1];
+        String eventEnd = components[2];
+        return new String[] {description, eventStart, eventEnd};
+    }
+
+    /**
+     * Parses string to get the find description.
+     * @param input The string input to parse
+     * @return String description to find tasks
+     */
+    public static String parseFindInfo (String input) throws EmptyDescriptionException {
+        String[] components = input.split("\\s+", 2);
+        // Missing Find Description.
+        if (components.length < 2 || components[1].equals("")) {
+            throw new EmptyDescriptionException("!!! Missing FIND Description !!!\n");
+        }
+        return components[1];
+    }
+
+
+    public Command parse(String input) {
+        try {
+            String[] parts = input.split(" ");
+
+            String command = parts[0].toLowerCase();
+            switch (command) {
+                case "bye":
+                    return new ByeCommand();
+                case "list":
+                    return new ListCommand();
+                case "delete":
+                    int indexToDelete = parseToGetIndex(input);
+                    return new DeleteCommand(indexToDelete);
+                case "todo":
+                    String toDoDescription = parseToDoInfo(input);
+                    return new ToDoCommand(toDoDescription);
+                case "deadline":
+                    String[] deadlineInfo = parseDeadlineInfo(input);
+                    return new DeadlineCommand(deadlineInfo[0], DateConverter.convertToDateTime(deadlineInfo[1]));
+                case "event":
+                    String[] eventInfo = parseEventInfo(input);
+                    return new EventCommand(eventInfo[0], DateConverter.convertToDateTime(eventInfo[1]),
+                            DateConverter.convertToDateTime(eventInfo[2]));
+                case "mark":
+                    int indexToMark = parseToGetIndex(input);
+                    return new MarkCommand(indexToMark);
+                case "unmark":
+                    int indexToUnmark = parseToGetIndex(input);
+                    return new UnmarkCommand(indexToUnmark);
+                case "find":
+                    String findDescription = parseFindInfo(input);
+                    return new FindCommand(findDescription);
+                default:
+                    return new InvalidCommand(input);
+            }
```

Comment:
Perhaps the switch/case/default statements could be at the same indentation according to the coding standard?